### PR TITLE
OPSAPS-51232. Optimized the mkdirs (not chowns)

### PR DESCRIPTION
**Pull request by ddraj**
Made the mkdirs optimal (one single command).

### CODE REVIEW

File: orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh [hunk 1]
```diff
@@ -38,6 +38,8 @@ EXISTING_USERS_FILE=/tmp/existing_users_file
 HDFS_KEYTAB=/run/cloudera-scm-agent/process/$(ls -t /run/cloudera-scm-agent/process/ | grep hdfs-NAMENODE$ | head -n 1)/hdfs.keytab
 echo "Keytab dir:$HDFS_KEYTAB"
 
+PATH_PREFIX="/user7"
```
Comment (keyki) [inline orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh]: Do we need this prefix?

File: orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh [hunk 2]
```diff
@@ -56,19 +58,26 @@ for existing in "${existingusers[@]}";
...
-     fi
+     echo "Will attempt to create HDFS home dir for $user"
+     newpaths+="$PATH_PREFIX/$user "
   fi
 done
-echo "End time of script $0, PID $$: `date`. Number of HDFS home dirs created $number_of_dirs_created."
+if [ -z "$newpaths" ]; then
+  echo "No home dirs to create in HDFS"
+else
+  hdfs dfs -mkdir -p $newpaths
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh]: will it handle hundreds of arguments? this should be tested on mow env before merged

File: orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh (hunk 1 above)
Comment (ddraj) [reply orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh]: Oh, the prefix should be set to /user .. I had set to user7 for testing and forgot to update.

File: orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh (hunk 2 above)
Comment (ddraj) [reply orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh]: Yes tested on mow-dev

**Comment by keyki** [comment]
I guess this can be closed, right?

**Pull request closed without merging**